src/platform/backends/lxd/lxd_virtual_machine.cpp

```diff
@@ -196,22 +196,29 @@ mp::LXDVirtualMachine::~LXDVirtualMachine()
 {
     update_shutdown_status = false;
 
-    current_state();
-    if (state == State::running)
+    try
```

Comment:
An overall wrapping with `top_catch_all` would still be nice (in addition to the existing try/catches), to account for any other exceptions that could be thrown here and avoid terminating in the same way.